src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp

```diff
@@ -124,13 +125,45 @@ QgsPointCloudRendererPropertiesWidget::QgsPointCloudRendererPropertiesWidget( Qg
   connect( mHorizontalTriangleThresholdSpinBox, qOverload<double>( &QgsDoubleSpinBox::valueChanged ), this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
   connect( mHorizontalTriangleUnitWidget, &QgsUnitSelectionWidget::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
 
-  // show label options only for virtual point clouds
-  bool showLabelOptions = !mLayer->dataProvider()->subIndexes().isEmpty();
-  mLabels->setVisible( showLabelOptions );
-  mLabelOptions->setVisible( showLabelOptions );
-  mLabelOptions->setDialogTitle( tr( "Customize label text" ) );
-  connect( mLabels, &QCheckBox::stateChanged, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
-  connect( mLabelOptions, &QgsFontButton::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+  // show virtual point cloud options only when vpc layer is selected
+  if ( !mLayer->dataProvider()->subIndexes().isEmpty() )
+  {
+    mLabelOptions->setDialogTitle( tr( "Customize label text" ) );
+    connect( mLabels, &QCheckBox::stateChanged, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+    connect( mLabelOptions, &QgsFontButton::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+    mZoomOutOptions->addItem( tr( "Show extents only" ), static_cast<int>( Qgis::PointCloudZoomOutBehavior::Extent ) );
+    try
+    {
+      const QgsVirtualPointCloudProvider &vpcProvider = dynamic_cast<QgsVirtualPointCloudProvider &>( *mLayer->dataProvider() );
+      if ( vpcProvider.overview() != nullptr )
+      {
+        mZoomOutOptions->addItem( tr( "Show overview only" ), static_cast<int>( Qgis::PointCloudZoomOutBehavior::Overview ) );
+        mZoomOutOptions->addItem( tr( "Show extents over overview" ), static_cast<int>( Qgis::PointCloudZoomOutBehavior::Both ) );
+      }
+    }
+    catch ( const std::bad_cast & )
+    {
+      mZoomOutOptions->setDisabled( true );
+    }
+    connect( mZoomOutOptions, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, [this]() {
+      Qgis::PointCloudZoomOutBehavior zoomOutBehavior = static_cast<Qgis::PointCloudZoomOutBehavior>( mZoomOutOptions->currentData().toInt() );
+      if ( zoomOutBehavior == Qgis::PointCloudZoomOutBehavior::Overview )
+      {
+        mLabels->setDisabled( true );
+        mLabelOptions->setDisabled( true );
+      }
+      else
+      {
+        mLabels->setDisabled( false );
+        mLabelOptions->setDisabled( false );
+      }
```

Comment:
1. A `switch` statement would be safer in case we decide to add other behaviors
2. `setDisabled( true )` disables input events. We'd probably prefer `setEnabled( false )` so the widgets also _look_ disabled.

Also below, in `syncToLayer()`

Reply:
I didn't know about `setEnabled()` and `setDisabled()` differences. `setDisabled()` just came to me naturally sa I wanted to disable the widget... It's also interesting that the widgets still had the _disabled_ look anyway.

I changed it in both positions and also to switch. To me it looks unnecessary now, but it looks cleaner.

Reply:
Whoops! You're right, `setEnabled()` and `setDisabled()` are just the same thing!

Regarding the `switch` statement though, the point is to _not_ have a default, so the compiler will complain if the enum gets new members!

Reply:
Fixed!